src/Task/ConvertTranslatableTask.php

```diff
@@ -0,0 +1,268 @@
+
+                foreach ($instances as $instance) {
+                    /** @var DataObject $instance */
+
+                    // Get the Locale column directly from the base table, since the SS ORM will not include it
+                    $instanceLocale = SQLSelect::create()
+                        ->setFrom($baseTable)
+                        ->setSelect('Locale')
+                        ->setWhere(['ID' => $instance->ID])
+                        ->execute()
+                        ->first();
+
+                    // Ensure that we got the Locale out of the base table before continuing
+                    if (empty($instanceLocale['Locale'])) {
+                        Debug::message("Skipping {$instance->Title} with ID {$instanceID} - couldn't find Locale");
+                        continue;
+                    }
+                    $instanceLocale = $instanceLocale['Locale'];
+
+                    // Check for obsolete classes that don't need to be handled any more
+                    if ($instance->ObsoleteClassName) {
+                        Debug::message("Skipping {$instance->ClassName} with ID {$instanceID} because it from an obsolete class", false);
+                        continue;
+                    }
+
+                    Debug::message(
+                        "Updating {$instance->ClassName} {$instance->MenuTitle} ({$instance->ID}) with locale {$instanceLocale}",
+                        false
+                    );
+
+                    FluentState::singleton()
+                        ->withState(function (FluentState $state) use ($instance, $instanceLocale) {
+                            // Use Fluent's ORM to write and/or publish the record into the correct locale
+                            // from Translatable
+                            $state->setLocale($instanceLocale);
+
+                            if (!$this->isPublished($instance)) {
+                                $instance->write();
+                                Debug::message("  --  Saved to draft", false);
+                            } elseif ($instance->publishRecursive() === false) {
+                                Debug::message("  --  Publishing FAILED", false);
+                                throw new Exception("Failed to publish");
+                            } else {
+                                Debug::message("  --  Published", false);
+                            }
+                        });
+                }
+
+                // Drop the "Locale" column from the base table
+                Debug::message('Dropping "Locale" column from ' . $baseTable, false);
+                DB::query(sprintf('ALTER TABLE "%s" DROP COLUMN "Locale"', Convert::raw2sql($baseTable)));
+
+                // Drop the "_translationgroups" translatable table
+                Debug::message('Deleting Translatable table ' . $groupTable, false);
+                DB::query(sprintf('DROP TABLE IF EXISTS "%s"', $groupTable));
```

Comment:
Again with the extreme edge cases, but `IF EXISTS` isn't always a thing either: https://blogs.msdn.microsoft.com/sqlserverstorageengine/2015/11/03/drop-if-exists-new-thing-in-sql-server-2016/

Reply:
See above, we don't support SQL Server